src/Microsoft.Fx.Portability.MetadataReader/DotNetFrameworkFilter.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Fx.Portability.ObjectModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Fx.Portability.Analyzer
+{
+    public class DotNetFrameworkFilter : IDependencyFilter
+    {
+        private static readonly ICollection<string> _microsoftKeys = new HashSet<string>(new[]
+        {
+            "b77a5c561934e089", // ECMA
+            "b03f5f7f11d50a3a", // DEVDIV
+            "7cec85d7bea7798e", // SLPLAT
+            "31bf3856ad364e35", // Windows
+            "24eec0d8c86cda1e", // Phone
+            "0738eb9f132ed756", // Mono
+            "ddd0da4d3e678217", // Component model
+            "84e04ff9cfb79065", // Mono Android
+            "842cf8be1de50553"  // Xamarin.iOS
+        }, StringComparer.OrdinalIgnoreCase);
```

Comment:
This is a list of all strong names in the %ProgramFiles%\Reference Assemblies\Microsoft directories. It catches all of the assemblies you listed.